### DISCUSSION TITLE
In the tutorial 5 of snnTorch the data that are inserted for training are the classic mnist and not spikes

### DISCUSSION THREAD

**Discussion by TheoTsio** [jeshraghian/snntorch#263] [General]
* snntorch version:
* Python version:
* Operating System:

### Description

train_loader = DataLoader(mnist_train, batch_size=batch_size, shuffle=True, drop_last=True)
test_loader = DataLoader(mnist_test, batch_size=batch_size, shuffle=True, drop_last=True)

It is using just this as input and there are no spikes

### What I Did

```
Paste the command(s) you ran and the output.
If there was a crash, please include the traceback here.
```

**Comment by jeshraghian** [comment]
The inputs don't have to be spikes. The idea is that the network will learn the optimal representation to convert non-spiking data into a format compatible with spiking. Tutorial 2 shows how non-spiking data is injected as a time-varying input current. The same logic applies.

**Reply by SM1991CODES** [reply]
Hi, while I get the idea, I don't understand - if the network learns the input spike representation, why do we still need time steps inside the forward function? Can it not be a single step and the network eventually learns generating correct spikes as it trains?

I was also wondering where is the time axis in the input MNIST frames and why don't we convert them to spikes before passing to the network? Is there a problem?
